### PR TITLE
📦 NEW: Check if BaseAI dev server is running

### PR DESCRIPTION
**Pull request by msaaddev**
This PR handles the following:

- Check if BaseAI dev server is running on port `9000`
  - Show user warning that the BaseAI dev server isn't running
- Check if port 9000 is in use and BaseAI dev server is running on it 
  - Show user warning that port is already in use and terminate the process and run the dev server

## Demo

image

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
 	},
 	"dependencies": {
 		"openai": "^4.63.0",
+		"tcp-port-used": "^1.0.2",
```
Comment (ahmadawais) [inline packages/core/package.json:36]: Unsure about this. @msaaddev adding this as a dependency could limit where the framework will work. Like if this `tcp-port-used` pkg isn't supported on Faslty or Workers, etc. I'd like there to not be a dependency like this.

File: packages/core/src/pipes/pipes.ts [hunk 2]
```diff
@@ -256,6 +261,38 @@ export class Pipe {
 		return currentResponse;
 	}
 
+	private async isLocalServerRunning(): Promise<Boolean> {
+		// TODO: Remove hard-coded port number when configurable support is added.
+		const portInUse = await isPortInUse(9000);
+
+		// Port not in use. BaseAI dev server is not running.
+		if (!portInUse) {
+			console.warn(
```
Comment (ahmadawais) [inline packages/core/src/pipes/pipes.ts:270]: think we need spacing, new line at the top and bottom of this. And it's an error, should be red.

File: packages/core/src/pipes/pipes.ts [hunk 3]
```diff
@@ -256,6 +261,38 @@ export class Pipe {
...
+				`\nBaseAI dev server is not running. Please run 'npx baseai@latest dev' to start the server.`,
```
Comment (ahmadawais) [inline packages/core/src/pipes/pipes.ts:271]:
```suggestion
`\nBaseAI dev server is not running. Please run 'npx baseai@latest dev’ in a new teriminal to start dev server.`,
```

File: packages/core/package.json (hunk 1 above)
Comment (ahmadawais) [reply packages/core/package.json:36]: Infact i think this is not going to work as this depends on node. We can't add that in core.

**Review by ahmadawais — CHANGES_REQUESTED** [review]

File: packages/core/package.json (hunk 1 above)
Comment (msaaddev) [reply packages/core/package.json:36]: Ah, makes sense. Let me see what else we can do about it.

Comment (msaaddev) [reply packages/core/package.json]: Removed the dependency.

**Review by ahmadawais — APPROVED** [review]
Looks good to me! ✔︎

**Pull request merged**